device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/auth/IotHubSasTokenAuthenticationProvider.java

```diff
@@ -34,7 +34,7 @@
     protected IotHubSasToken sasToken;
 
     public abstract boolean canRefreshToken();
-    public abstract String getRenewedSasToken(boolean proactivelyRenew, boolean forceRenewal) throws IOException, TransportException;
+    public abstract char[] getSasToken() throws IOException, TransportException;
```

Comment:
This class is a public class yes? can we just change the method ?

Reply:
No, these AuthenticationProvider class are not part of our public API surface. You can't construct a device client with one of these things